src/Datasource/Connection.php

```diff
@@ -330,7 +332,7 @@ public function prepare($sql)
      * @see \Cake\Datasource\ConnectionInterface::getDriver()
      * @return \Elastica\Client
      */
-    public function getDriver()
+    public function getDriver(): object
```

Comment:
```suggestion
    public function getDriver(): Client
```

This should be able to narrow the return type.